### ISSUE TITLE
[Question] Role Manager with Resource Role

### ISSUE THREAD

**Issue by ChenHaolinOlym** [casbin/casbin#1075]
**Want to prioritize this issue? Try:**

[![issuehunt-to-marktext](https://github.com/BoostIO/issuehunt-materials/raw/master/v1/issuehunt-button-v1.svg)](https://issuehunt.io/r/casbin/casbin)

------

**What's your scenario? What do you want to achieve?**
I'm working with a resource role rbac model, so I need apis such as `e.GetResourcesForResourceRole()` and `e.GetResourceRolesForResource()`.
I check their counterparts for roles and users, they are implemented with role managers.
But after skim through the source code, I found no way of getting the role manager for resource roles. `rmMap` is not exported and `GetRoleManager` only returns the one for users and roles.
So I'm here to ask if there's another way to get the role manager for resource roles.

**Your model:**

```ini
[request_definition]
r = sub, obj, act

[policy_definition]
p = sub, obj, act

[role_definition]
g = _, _
g2 = _, _

[policy_effect]
e = some(where (p.eft == allow))

[matchers]
m = g(r.sub, p.sub) && g2(r.obj, p.obj) && regexMatch(r.act, p.act)
```

**Comment by hsluoyz** [comment]
@JalinWang 

/cc @tangyang9464

**Comment by JalinWang** [comment]
@tangyang9464 
I think `GetRoleManager` & `SetRoleManager` should support fetching `g*` section other than the default one, e.g., `g2`.
```go
// GetRoleManager gets the current role manager.
func (e *Enforcer) GetRoleManager() rbac.RoleManager {
	return e.rmMap["g"]
}

// SetRoleManager sets the current role manager.
func (e *Enforcer) SetRoleManager(rm rbac.RoleManager) {
	e.rmMap["g"] = rm
}
```

Could we modify the interface?
```go
// IEnforcer is the API interface of Enforcer
type IEnforcer interface {
...
	GetRoleManager() rbac.RoleManager
	SetRoleManager(rm rbac.RoleManager)
...
}
```

**Comment by tangyang9464** [comment]
> @tangyang9464 
> I think `GetRoleManager` & `SetRoleManager` should support fetching `g*` section other than the default one, e.g., `g2`.
> ```go
> // GetRoleManager gets the current role manager.
> func (e *Enforcer) GetRoleManager() rbac.RoleManager {
> 	return e.rmMap["g"]
> }
> 
> // SetRoleManager sets the current role manager.
> func (e *Enforcer) SetRoleManager(rm rbac.RoleManager) {
> 	e.rmMap["g"] = rm
> }
> ```
> 
> Could we modify the interface?
> ```go
> // IEnforcer is the API interface of Enforcer
> type IEnforcer interface {
> ...
> 	GetRoleManager() rbac.RoleManager
> 	SetRoleManager(rm rbac.RoleManager)
> ...
> }
> ```

@JalinWang Implemented in #1040. You can open a PR alone to realize it
